From a0ecfa0d5c8dfc961b896466749decdb5bb6422b Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Wed, 29 Oct 2008 19:50:24 +1100
Subject: [PATCH] e-wm: update efl svnrev and change e package to reflect upstream changes

upstream has moved illume into the e core tree and added first-run wizard
support so now you just choose your favorite setup at runtime. all themes
etc. are now merged and you save a lot of flash space.
---
 conf/distro/include/sane-srcrevs.inc |    4 +-
 packages/e17/e-wm_svn.bb             |   85 +++++++++++++++++++++++++++++----
 2 files changed, 75 insertions(+), 14 deletions(-)

diff --git a/conf/distro/include/sane-srcrevs.inc b/conf/distro/include/sane-srcrevs.inc
index d566c8e..c42d47f 100644
--- a/conf/distro/include/sane-srcrevs.inc
+++ b/conf/distro/include/sane-srcrevs.inc
@@ -196,7 +196,7 @@ SRCREV_pn-zhone ?= "28e01227063ab76b9144e7a443319add6b195864"
 # Enlightenment Foundation Libraries
 # Caution: This is not alphabetically, but (roughly) dependency-sorted.
 # Please leave it like that.
-EFL_SRCREV ?= "36882"
+EFL_SRCREV ?= "37299"
 SRCREV_pn-edb-native ?= "${EFL_SRCREV}"
 SRCREV_pn-edb ?= "${EFL_SRCREV}"
 SRCREV_pn-eina-native ?= "${EFL_SRCREV}"
@@ -259,8 +259,6 @@ SRCREV_pn-enna ?= "${EFL_SRCREV}"
 SRCREV_pn-exquisite ?= "${EFL_SRCREV}"
 SRCREV_pn-exquisite-theme-illume ?= "${EFL_SRCREV}"
 SRCREV_pn-rage ?= "${EFL_SRCREV}"
-SRCREV_pn-illume ?= "${EFL_SRCREV}"
-SRCREV_pn-illume-theme-illume ?= "${EFL_SRCREV}"
 
 # Enlightenment ST
 SRCREV_pn-alarm ?= "${EFL_SRCREV}"
diff --git a/packages/e17/e-wm_svn.bb b/packages/e17/e-wm_svn.bb
index 4309500..0f67234 100644
--- a/packages/e17/e-wm_svn.bb
+++ b/packages/e17/e-wm_svn.bb
@@ -2,11 +2,51 @@ DESCRIPTION = "The Enlightenment Window Mananger Version 17"
 DEPENDS = "eet evas ecore edje efreet edbus"
 LICENSE = "MIT BSD"
 PV = "0.16.999.043+svnr${SRCREV}"
-FILE_PR = "r15"
+FILE_PR = "r19"
 
 inherit e update-alternatives
 
-RDEPENDS += "shared-mime-info mime-support edje-utils"
+RDEPENDS += "shared-mime-info mime-support edje-utils glibc-utils"
+
+PACKAGES =+ "\
+  ${PN}-config-default \
+  ${PN}-config-illume \
+  ${PN}-config-minimalist \
+  ${PN}-config-netbook \
+  ${PN}-config-scaleable \
+  ${PN}-config-standard \
+  ${PN}-theme-default \
+  ${PN}-theme-illume \
+  ${PN}-background-dark-gradient \
+  ${PN}-background-light-gradient \
+  ${PN}-images \
+  ${PN}-icons \
+  ${PN}-other \
+  ${PN}-input-methods \
+"
+
+RRECOMMENDS_${PN} = "\
+  ${PN}-config-default \
+  ${PN}-images \
+  ${PN}-icons \
+  ${PN}-other \
+  ${PN}-input-methods \
+"
+
+PACKAGE_ARCH_${PN}-config-default = "all"
+PACKAGE_ARCH_${PN}-config-illume = "all"
+PACKAGE_ARCH_${PN}-config-minimalist = "all"
+PACKAGE_ARCH_${PN}-config-netbook = "all"
+PACKAGE_ARCH_${PN}-config-scaleable = "all"
+PACKAGE_ARCH_${PN}-config-standard = "all"
+PACKAGE_ARCH_${PN}-theme-default = "all"
+PACKAGE_ARCH_${PN}-theme-illume = "all"
+PACKAGE_ARCH_${PN}-background-dark-gradient = "all"
+PACKAGE_ARCH_${PN}-background-light-gradient = "all"
+PACKAGE_ARCH_${PN}-images = "all"
+PACKAGE_ARCH_${PN}-icons = "all"
+PACKAGE_ARCH_${PN}-other = "all"
+PACKAGE_ARCH_${PN}-input-methods = "all"
 
 SRC_URI = "\
   svn://svn.enlightenment.org/svn/e/trunk;module=e;proto=http \
@@ -37,30 +77,53 @@ do_stage() {
 
 do_install_append() {
     # customising - should rather make this simple upstream
-    rm -rf ${D}${datadir}/enlightenment/data/backgrounds/*
-    rm -rf ${D}${datadir}/enlightenment/data/fonts/*
-    rm -rf ${D}${datadir}/enlightenment/data/other/*
     install -m 755 ${WORKDIR}/enlightenment_start.oe ${D}/${bindir}
     install -d ${D}/${sysconfdir}/xdg/menus
     install -m 644 ${WORKDIR}/applications.menu ${D}/${sysconfdir}/xdg/menus/
+    for I in `find ${D}/${libdir}/enlightenment -name "*.a" -print`; do rm -f $I; done
+    for I in `find ${D}/${libdir}/enlightenment -name "*.la" -print`; do rm -f $I; done
 }
 
 FILES_${PN} = "\
   ${bindir}/* \
   ${libdir}/enlightenment/modules/*/*.* \
   ${libdir}/enlightenment/modules/*/*/* \
-  ${libdir}/enlightenment/preload/*.so \
-  ${datadir} \
+  ${datadir}/locale \
+  ${datadir}/enlightenment/data/icons \
+  ${datadir}/enlightenment/data/input_methods \
+  ${datadir}/enlightenment/data/config/profile.cfg \
   ${sysconfdir} \
 "
+FILES_${PN}-config-default = "${datadir}/enlightenment/data/config/default"
+FILES_${PN}-config-illume = "${datadir}/enlightenment/data/config/illume"
+FILES_${PN}-config-minimalist = "${datadir}/enlightenment/data/config/minimalist"
+FILES_${PN}-config-netbook = "${datadir}/enlightenment/data/config/netbook"
+FILES_${PN}-config-scaleable = "${datadir}/enlightenment/data/config/scaleable"
+FILES_${PN}-config-standard = "${datadir}/enlightenment/data/config/standard"
+FILES_${PN}-theme-default = "${datadir}/enlightenment/data/themes/default.edj"
+FILES_${PN}-theme-illume = "${datadir}/enlightenment/data/themes/illume.edj"
+FILES_${PN}-theme-default = "${datadir}/enlightenment/data/themes/default.edj"
+FILES_${PN}-background-dark-gradient = "${datadir}/enlightenment/data/backgrounds/Dark_Gradient.edj"
+FILES_${PN}-background-light-gradient = "${datadir}/enlightenment/data/backgrounds/Light_Gradient.edj"
+FILES_${PN}-images = "${datadir}/enlightenment/data/images"
+FILES_${PN}-icons = "${datadir}/enlightenment/data/icons"
+FILES_${PN}-other = "${datadir}/enlightenment/data/other"
+FILES_${PN}-input-methods = "${datadir}/enlightenment/data/input_methods"
+
+RRECOMMENDS_${PN}-config-default = "${PN}-theme-default"
+RRECOMMENDS_${PN}-config-illume = "${PN}-theme-illume"
+RRECOMMENDS_${PN}-config-minimalist = "${PN}-background-light-gradient \
+${PN}-theme-default"
+RRECOMMENDS_${PN}-config-netbook = "${PN}-background-dark-gradient \
+${PN}-theme-default"
+RRECOMMENDS_${PN}-config-scaleable = "${PN}-theme-default"
+RRECOMMENDS_${PN}-config-standard = "${PN}-theme-default"
+
 FILES_${PN}-dbg += "\
   ${libdir}/enlightenment/modules/*/*/.debug/ \
   ${libdir}/enlightenment/preload/.debug/ \
 "
-
-FILES_${PN}-dev += "\
-  ${libdir}/enlightenment/preload/*.?a \
-"
+FILES_${PN}-dev += "${libdir}/enlightenment/preload/*.?a"
 
 CONFFILES_${PN} = "/etc/xdg/menus/applications.menu"
 
-- 
1.5.4.3

